--- SDL_windowsevents.c.orig	2014-03-16 00:01:43.000000000 -0230
+++ SDL_windowsevents.c	2014-09-07 17:51:51.983290900 -0230
@@ -765,19 +765,23 @@
         }
         return (1);
 
-#if defined(SC_SCREENSAVE) || defined(SC_MONITORPOWER)
     case WM_SYSCOMMAND:
         {
+            /* Disable beep on Alt-key press */
+            if ((wParam & 0xFFF0) == SC_KEYMENU) {
+                return (0);
+            }
+#if defined(SC_SCREENSAVE) || defined(SC_MONITORPOWER)
             /* Don't start the screensaver or blank the monitor in fullscreen apps */
-            if ((wParam & 0xFFF0) == SC_SCREENSAVE ||
+            else if ((wParam & 0xFFF0) == SC_SCREENSAVE ||
                 (wParam & 0xFFF0) == SC_MONITORPOWER) {
                 if (SDL_GetVideoDevice()->suspend_screensaver) {
                     return (0);
                 }
             }
+#endif /* System has screensaver support */
         }
         break;
-#endif /* System has screensaver support */
 
     case WM_CLOSE:
         {
